moveit_plugins/moveit_simple_controller_manager/include/moveit_simple_controller_manager/action_based_controller_handle.h

```diff
@@ -123,6 +129,7 @@ class ActionBasedControllerHandle : public ActionBasedControllerHandleBase
     {
       RCLCPP_INFO_STREAM(LOGGER, "Cancelling execution for " << name_);
       auto cancel_result_future = controller_action_client_->async_cancel_goal(current_goal_);
+      //@todo: replace rclcpp::spin_until_future_complete
       if (rclcpp::spin_until_future_complete(node_, cancel_result_future) !=
           rclcpp::executor::FutureReturnCode::SUCCESS)
```

Comment:
So, for `std::shared_future` results from async calls we should just use `wait()` `wait_for()` or `wait_until()` since the executor is already doing the spinning.